### ISSUE TITLE
Application unexpectedly terminated: File or Assembly not found

### ISSUE THREAD

**Issue by binderth** [phw198/OutlookGoogleCalendarSync#434]
----

**OGCS Version**: v2.6.6.0-alpha
**Installed or Portable**: Installed

**Problem Description**  
While starting OutlookGoogleCalendarSync, I get immediately the following error-alert:
![image](https://user-images.githubusercontent.com/9037329/35028347-ed9b7e4a-fb55-11e7-9a6b-ca00abfd1efd.png)

**Steps to Reproduce the Issue**  
1. start OGCS

**Any other information**  
Contents of log:
```
2018-01-17 00:29:49,571 INFO  1 OutlookGoogleCalendarSync.Program [0] -  Program started: v2.6.6.0
2018-01-17 00:29:49,571 INFO  1 OutlookGoogleCalendarSync.Program [0] -  Started interactively.
2018-01-17 00:29:49,571 INFO  1 OutlookGoogleCalendarSync.Program [0] -  Storing user files in roaming directory: C:\Users\thomas.binder\AppData\Roaming\Outlook Google Calendar Sync
2018-01-17 00:29:49,571 INFO  1 OutlookGoogleCalendarSync.Program [0] -  Running from C:\Users\thomas.binder\AppData\Local\OutlookGoogleCalendarSync\app-2.6.6-alpha\OutlookGoogleCalendarSync.exe
2018-01-17 00:29:49,587 DEBUG  1 OutlookGoogleCalendarSync.XMLManager [0] -  Retrieved setting 'Portable' with value 'false'
2018-01-17 00:29:49,587 DEBUG  1 OutlookGoogleCalendarSync.XMLManager [0] -  Retrieved setting 'LoggingLevel' with value 'DEBUG'
2018-01-17 00:29:49,587 INFO  1 OutlookGoogleCalendarSync.Settings [0] -  Logging level configured to 'DEBUG'
2018-01-17 00:29:49,587 INFO  1 OutlookGoogleCalendarSync.Program [0] -  Purging log files older than 30 days...
2018-01-17 00:29:49,587 INFO  1 OutlookGoogleCalendarSync.Program [0] -  Purge complete.
2018-01-17 00:29:49,587 DEBUG  1 OutlookGoogleCalendarSync.Updater [0] -  Setting up Squirrel handlers.
2018-01-17 00:29:49,805 DEBUG  3 OutlookGoogleCalendarSync.XMLManager [0] -  Retrieved setting 'CompletedSyncs' with value '1828'
2018-01-17 00:29:49,805 DEBUG  3 OutlookGoogleCalendarSync.XMLManager [0] -  Retrieved setting 'Donor' with value 'false'
2018-01-17 00:29:49,805 DEBUG  3 OutlookGoogleCalendarSync.XMLManager [0] -  Retrieved setting 'Subscribed' with value '2000-01-01T00:00:00'
2018-01-17 00:29:49,805 DEBUG  3 OutlookGoogleCalendarSync.XMLManager [0] -  Retrieved setting 'HideSplashScreen' with value 'false'
2018-01-17 00:29:49,821 DEBUG  3 OutlookGoogleCalendarSync.Splash [0] -  Showing splash screen.
2018-01-17 00:29:49,852 DEBUG  1 OutlookGoogleCalendarSync.Program [0] -  Loading settings from file.
2018-01-17 00:29:49,930 INFO  1 OutlookGoogleCalendarSync.Updater [0] -  This is a Squirrel install.
2018-01-17 00:29:49,930 DEBUG  1 OutlookGoogleCalendarSync.Updater [0] -  Automatic update check requested.
2018-01-17 00:29:49,930 INFO  1 OutlookGoogleCalendarSync.SettingsProxy [0] -  Using default proxy (app.config / IE).
2018-01-17 00:29:49,930 INFO  1 OutlookGoogleCalendarSync.SettingsProxy [0] -  Setting system-wide proxy.
2018-01-17 00:29:49,946 DEBUG  1 OutlookGoogleCalendarSync.SettingsProxy [0] -  Testing the system proxy.
2018-01-17 00:29:49,946 DEBUG  1 OutlookGoogleCalendarSync.SettingsProxy [0] -  Confirmation of configured proxy: http://www.google.com
2018-01-17 00:29:49,962 DEBUG  1 OutlookGoogleCalendarSync.Updater [0] -  Checking for Squirrel update...
2018-01-17 00:29:50,009 INFO  1 OutlookGoogleCalendarSync.NotificationTray [0] -  Using NodaTime TZDB: 2017c (mapping: 13290)
2018-01-17 00:29:50,009 DEBUG  10 OutlookGoogleCalendarSync.NotificationTray [0] -  Checking for new timezone database...
2018-01-17 00:29:50,009 DEBUG  1 OutlookGoogleCalendarSync.MainForm [0] -  Initialiasing MainForm.
2018-01-17 00:29:50,509 DEBUG  1 OutlookGoogleCalendarSync.GoogleOgcs.Authenticator [0] -  Getting MD5 hash for '*******@gmail.com'
2018-01-17 00:29:50,524 ERROR  1 OutlookGoogleCalendarSync.Social [0] -  Failed setting Outlook client analytics URL.
2018-01-17 00:29:50,524 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  System.IO.FileNotFoundException: Die Datei oder Assembly "office, Version=12.0.0.0, Culture=neutral, PublicKeyToken=******e111******" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden.
2018-01-17 00:29:50,524 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x80070002;-2147024894
2018-01-17 00:29:50,524 DEBUG  1 OutlookGoogleCalendarSync.Social [0] -  Retrieving URL: https://phw198.github.io/OutlookGoogleCalendarSync/track/ogcs?version=Unknown
2018-01-17 00:29:50,540 DEBUG  1 OutlookGoogleCalendarSync.Social [0] -  Retrieving URL: https://www.google-analytics.com/collect?v=1&t=event&tid=UA-19426033-4&cid=203d41909f09da30f96cc4a1c0756161&ea=version&ec=ogcs&el=2.6.6.0
2018-01-17 00:29:50,571 DEBUG  1 OutlookGoogleCalendarSync.MainForm [0] -  Find Folders
2018-01-17 00:29:50,571 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  System.IO.FileNotFoundException: Die Datei oder Assembly "office, Version=12.0.0.0, Culture=neutral, PublicKeyToken=******e111******" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden.
2018-01-17 00:29:50,571 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x80070002;-2147024894
2018-01-17 00:29:50,571 INFO  1 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [0] -  It appears Outlook has been restarted after OGCS was started. Reconnecting...
2018-01-17 00:29:50,571 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  System.IO.FileNotFoundException: Die Datei oder Assembly "office, Version=12.0.0.0, Culture=neutral, PublicKeyToken=******e111******" oder eine Abhängigkeit davon wurde nicht gefunden. Das System kann die angegebene Datei nicht finden.
2018-01-17 00:29:50,571 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x80070002;-2147024894
2018-01-17 00:29:50,571 FATAL  1 OutlookGoogleCalendarSync.Program [0] -  Application unexpectedly terminated!
2018-01-17 00:29:50,883 DEBUG  10 OutlookGoogleCalendarSync.NotificationTray [0] -  Already have latest TZDB version.
2018-01-17 00:29:51,963 INFO  7 OutlookGoogleCalendarSync.Updater [0] -  Already running the latest version of OGCS.
2018-01-17 00:29:58,002 DEBUG  3 OutlookGoogleCalendarSync.Splash [0] -  Disposed of splash screen.
2018-01-17 00:30:18,556 WARN  1 OutlookGoogleCalendarSync.Program [0] -  Tidying down any remaining Outlook references, as OGCS crashed out.
2018-01-17 00:30:18,556 INFO  1 OutlookGoogleCalendarSync.Program [0] -  Application closed.

**Comment by phw198** [comment]
Apologies, will have to investigate this further when I get a moment. Can you confirm how you installed? Was it an in-situ upgrade or fresh install from `Setup.exe`?

In the mean time, can you try [extracting this file](https://github.com/phw198/OutlookGoogleCalendarSync/files/1639147/Microsoft.Office.Interop.Outlook.zip) into the same directory as the OGCS executable.

**Comment by binderth** [comment]
no apologies needed... ;)

It was an upgrade from the application itself.
The provided DLL already was inside the directory. replacing it didn't change anything unfortunately.

I just deinstalled an re-installed anew - same error

**Comment by phw198** [comment]
I've just tested upgrading to 2.6.6 again it works OK for me. There are others who have upgraded OK too, so it looks like a problem with your system. 

Did the problem start as soon as you upgraded to 2.6.6? If not, maybe Office was patched and it's messed up your Outlook install? Could try a Repair?

Have you tried downgrading back to 2.6.5? I'm assuming that version won't work either now...

I think the key part of the error message is `oder eine Abhängigkeit davon wurde nicht gefunden.` - a missing dependency that should be there if Outlook is installed. A [similar issue](https://social.msdn.microsoft.com/Forums/sqlserver/en-US/ef7b2e05-4f0c-4b98-a4bc-92d05dbc7194/systemiofilenotfoundexception-file-or-assembly-name-xxxxx-or-one-of-its-dependencies-was-not?forum=vsdebug) was reportedly fixed by using Dependency Walker (I've never used this) to find the missing dependency and then add the path to it to your %PATH% system environment. Worth a shot?

Also, check if any of the files in the OGCS directory have been blocked as coming from an "untrusted source" - check the properties and `Unblock` if they have.

**Comment by binderth** [comment]
I do think, the problem started some days ago. I don't remember having it immediately after the update. I think, that was already here befort 2.6.6. 
What was happening, that Windows did an update (I don't think, Office). The updates were .NET related (Microsoft .NET Framework 4.7.1 für Windows 10 Version 1607 für x64 (KB4033393) on January 11th). That date could be the one since the error came. I clicked something (and I let my Laptop in deep sleep mostly). Yesterday I booted it and saw the error clearly.

adding **"C:\Users\thomas.binder\AppData\Local\OutlookGoogleCalendarSync\app-2.6.6-alpha"** to the %PATH% didn't help either...

What I discovered (but long ago and didn't interfere with OGCS since!) is, that my Win10 can't install the Fall Creators Update. I'm trying since to get it done, but it always reverts.
